features/google/src/model/importers/semantic-locations-importer.js

```diff
@@ -0,0 +1,89 @@
+import ActivitySegment from "../entities/activity-segment";
+import PlaceVisit from "../entities/place-visit";
+
+const semanticLocationsRegex =
+    /\/Location History\/Semantic Location History\/.*\.json$/;
+
+async function readFullPathJSONFile(entry) {
+    const rawContent = await entry.getContent();
+    const fileContent = new TextDecoder("utf-8").decode(rawContent);
+
+    return JSON.parse(fileContent);
+}
+
+/**
+ * We saw until now two different formats for timestamps:
+ * - milliseconds: 1394270917000
+ * - standard date: "2022-01-19T14:28:16.967Z"
+ */
+function extractTimestampFromDuration(duration) {
+    if ("startTimestamp" in duration) return new Date(duration.startTimestamp);
+    if ("startTimestampMs" in duration)
+        return new Date(duration.startTimestampMs);
+    throw new Error("Unknown start date");
```

Comment:
Please make it more informative by adding `duration`
Also, this is kind of brittle. Have you considered looking for the first attribute that includes `Timestamp` and feeding it into a `Date`?

Reply:
You mean adding `duration` as an attribute to `PlaceVisit`?

Until now we only saw those two types of formats. There are also the corresponding `endTimestamp` and `endTimestampMs`. I find having the explicit list as reducing the number of surprizes that we can have. We never know if what we pass to the Date constructor will work, if we did not see that data before.

Reply:
No, to the error. Sorry, I'll clarify that.